fix(canon): relative typeRoots inherited via tsconfig extends are mangled to node_modules/node_modules/@types → false TS2688 masks all file diagnostics

### Area

CLI

### Version

vize 0.347.7 (npm package, `@typescript/native-preview` 7.0.0-dev.20260603.1), macOS arm64, pnpm

### Summary

When a package's `tsconfig.json` inherits a **relative `typeRoots`** via `extends` from a shared-tsconfig workspace package, `vize check` rewrites the paths into the materialized Corsa project incorrectly — they come out as `./node_modules/node_modules/@types` (note the doubled `node_modules`), which does not exist. The result is a false `TS2688 Cannot find type definition file for '<lib>'` for every entry in `types`.

Per TypeScript semantics, relative paths in an extended config resolve relative to **the config file that defines them**, and `tsc` / `vue-tsc` accept this setup with no diagnostics.

The impact is worse than one false error: because TypeScript treats a failed type-library load as a blocking global diagnostic (verified: plain `tsc` behaves the same way once the config is genuinely broken), the mangled config **suppresses every per-file diagnostic**. On a real project where we evaluated vize, a file seeded with 5 obvious type errors (template + script) came back with `"diagnostics": []` while only the 2 false TS2688s were reported — so the run fails, but for the wrong reason, and none of the real errors are visible. `vue-tsc` on the identical project reports all seeded errors and no TS2688.

### Minimal reproduction

Self-contained script (only public npm packages):

```bash
mkdir vize-typeroots-repro && cd vize-typeroots-repro
mkdir -p packages/shared-tsconfig/base apps/app/src

cat > package.json << 'EOF'
{
  "name": "vize-typeroots-repro",
  "private": true,
  "devDependencies": {
    "@types/node": "22.18.8",
    "typescript": "5.9.3",
    "vize": "0.347.7",
    "vue-tsc": "3.0.8"
  }
}
EOF

cat > pnpm-workspace.yaml << 'EOF'
packages:
    - 'packages/*'
    - 'apps/*'
EOF

cat > packages/shared-tsconfig/package.json << 'EOF'
{ "name": "@repro/shared-tsconfig", "version": "1.0.0", "private": true }
EOF

# typeRoots is relative to THIS file (packages/shared-tsconfig/base/):
# ../../../node_modules/@types = <workspace root>/node_modules/@types
cat > packages/shared-tsconfig/base/tsconfig.json << 'EOF'
{
  "compilerOptions": {
    "target": "es2020",
    "module": "ESNext",
    "moduleResolution": "bundler",
    "strict": true,
    "skipLibCheck": true,
    "typeRoots": ["../../../node_modules/@types"],
    "types": ["node"]
  }
}
EOF

cat > apps/app/package.json << 'EOF'
{
  "name": "@repro/app",
  "version": "1.0.0",
  "private": true,
  "dependencies": { "vue": "3.4.23" },
  "devDependencies": { "@repro/shared-tsconfig": "workspace:*" }
}
EOF

cat > apps/app/tsconfig.json << 'EOF'
{
  "extends": "@repro/shared-tsconfig/base/tsconfig.json",
  "include": ["src/**/*"]
}
EOF

cat > apps/app/src/broken.vue << 'EOF'
<template>
    <span>{{ undefinedProperty }}</span>
</template>
<script setup lang="ts">
const count: number = 'this is a string'
</script>
EOF

pnpm install
cd apps/app
pnpm exec vue-tsc --noEmit          # 2 real errors, no TS2688
pnpm exec vize check --format json  # 1 false TS2688, broken.vue: []
```

### Actual behavior

`vue-tsc --noEmit` (correct baseline):

```
src/broken.vue(2,14): error TS2339: Property 'undefinedProperty' does not exist on type ...
src/broken.vue(5,7): error TS2322: Type 'string' is not assignable to type 'number'.
```

`vize check --format json`:

```json
{
  "files": [
    { "file": "src/broken.vue", "diagnostics": [] },
    { "file": "tsconfig.json", "diagnostics": [
      "error:1:1 [TS2688] Cannot find type definition file for 'node'. ..."
    ] }
  ],
  "errorCount": 1
}
```

The materialized project under `node_modules/.vize/canon/projects/<hash>/tsconfig.json` shows the mangled paths:

```json
"typeRoots": [
  "./node_modules/node_modules/@types",
  "../../../../../node_modules/node_modules/@types"
]
```

Note: the generated virtual TS for `broken.vue` is correct and contains both seeded errors — generation is fine; the diagnostics disappear because of the broken (mangled) config, consistent with how `tsc` treats a failing type-library load.

Control experiment: declaring the **same** `typeRoots` directly in the leaf `apps/app/tsconfig.json` (as `../../node_modules/@types`, i.e. relative to the app) materializes correctly (an absolute path to the workspace root `node_modules/@types`) and all seeded diagnostics come back. Only the `extends`-inherited relative form is mangled.

### Expected behavior

Relative `typeRoots` (and other relative paths) declared in an extended tsconfig should resolve relative to the defining config file, matching `tsc` / `vue-tsc`: no TS2688, and the two seeded errors in `broken.vue` are reported.

--- a/crates/vize_canon/src/batch/virtual_project/tests/tsconfig_extends.rs
+++ b/crates/vize_canon/src/batch/virtual_project/tests/tsconfig_extends.rs
@@ -220,3 +220,102 @@ fn external_snapshot_is_flattened_and_cache_location_independent() {
 
     let _ = fs::remove_dir_all(&case_dir);
 }
+
+/// A bare `extends` specifier resolves through Node resolution, which follows
+/// symlinks, so the relative options the extended config declares anchor to the
+/// config's real directory rather than to the `node_modules` entry that pointed
+/// at it. Under pnpm every workspace dependency is such a link, and anchoring to
+/// the link walked `../..` back out through `node_modules` and produced a
+/// `node_modules/node_modules/@types` root that cannot exist — a TS2688 that
+/// suppresses every per-file diagnostic in the run (#4425).
+#[cfg(unix)]
+#[test]
+fn relative_type_roots_from_a_linked_package_anchor_to_the_real_directory() {
+    let case_dir = unique_case_dir("tsconfig-linked-type-roots");
+    let _ = fs::remove_dir_all(&case_dir);
+    let root = case_dir.join("workspace");
+    let shared = root.join("packages/shared-tsconfig/base");
+    let app = root.join("apps/app");
+    fs::create_dir_all(&shared).unwrap();
+    fs::create_dir_all(app.join("src")).unwrap();
+    fs::create_dir_all(root.join("node_modules/@types/node")).unwrap();
+    fs::write(
+        root.join("node_modules/@types/node/index.d.ts"),
+        "export {};",
+    )
+    .unwrap();
+
+    // `../../../node_modules/@types` from the shared package's own directory is
+    // the workspace root's `@types`.
+    fs::write(
+        shared.join("tsconfig.json"),
+        r#"{
+  "compilerOptions": {
+    "strict": true,
+    "typeRoots": ["../../../node_modules/@types"],
+    "types": ["node"]
+  }
+}"#,
+    )
+    .unwrap();
+
+    // pnpm's shape: the dependency is a link into the workspace, not a copy.
+    let app_modules = app.join("node_modules/@repro");
+    fs::create_dir_all(&app_modules).unwrap();
+    std::os::unix::fs::symlink(
+        root.join("packages/shared-tsconfig"),
+        app_modules.join("shared-tsconfig"),
+    )
+    .unwrap();
+
+    fs::write(
+        app.join("tsconfig.json"),
+        r#"{
+  "extends": "@repro/shared-tsconfig/base/tsconfig.json",
+  "include": ["src/**/*"]
+}"#,
+    )
+    .unwrap();
+    let vue_path = app.join("src/App.vue");
+    fs::write(
+        &vue_path,
+        "<script setup lang=\"ts\">const count = 1</script>",
+    )
+    .unwrap();
+
+    let mut project = VirtualProject::new(&app).unwrap();
+    project.register_path(&vue_path).unwrap();
+    project.materialize().unwrap();
+
+    let tsconfig_path = project.virtual_root().join("tsconfig.json");
+    let value: serde_json::Value =
+        serde_json::from_str(&fs::read_to_string(tsconfig_path).unwrap()).unwrap();
+    let type_roots = value["compilerOptions"]["typeRoots"]
+        .as_array()
+        .unwrap()
+        .iter()
+        .map(|entry| entry.as_str().unwrap().to_owned())
+        .collect::<Vec<_>>();
+
+    for entry in &type_roots {
+        assert!(
+            !entry.contains("node_modules/node_modules"),
+            "typeRoots must not walk back out through node_modules: {type_roots:?}",
+        );
+    }
+    // The root sits outside the app, so it stays absolute and is listed once —
+    // `remap_dir_entries` only pairs a mirror copy with a real-tree fallback for
+    // entries that live under the project root.
+    let expected = root.join("node_modules/@types");
+    assert_eq!(
+        type_roots,
+        vec![
+            vize_carton::path::canonicalize_non_verbatim(&expected)
+                .to_string_lossy()
+                .into_owned()
+        ],
+        "expected the workspace root's @types",
+    );
+
+    let _ = fs::remove_dir_all(&case_dir);
+}

--- a/crates/vize_canon/src/batch/virtual_project/tsconfig_paths.rs
+++ b/crates/vize_canon/src/batch/virtual_project/tsconfig_paths.rs
@@ -68,7 +68,18 @@ fn resolve_package_tsconfig_path(base_dir: &Path, extends: &str) -> Option<PathB
             .map(|candidate| normalize_path_lexically(&candidate))
             .find(|candidate| candidate.is_file())
         {
-            return Some(found);
+            // Resolve through the link before returning. A bare specifier goes
+            // through Node resolution, which follows symlinks unless
+            // `preserveSymlinks` is set, so the config's own directory — the
+            // anchor for every relative option it declares — is the real one,
+            // not the `node_modules` entry that pointed at it. Under pnpm those
+            // differ for every workspace package: rebasing a relative
+            // `typeRoots` against `<pkg>/node_modules/<dep>/...` walked back out
+            // through `node_modules` and produced `node_modules/node_modules/
+            // @types`, a directory that cannot exist, and the resulting TS2688
+            // is a global failure that suppresses every per-file diagnostic
+            // (#4425).
+            return Some(vize_carton::path::canonicalize_non_verbatim(&found));
         }
         current = dir.parent();
     }

